docs/tutorials/web/large-models.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Working with Large Models
+description: Working with Large Models in ONNX Runtime Web
+parent: Web
+grand_parent: Tutorials
+has_children: false
+nav_order: 3
+---
+{::options toc_levels="2..4" /}
+
+# Working with Large Models
+{: .no_toc }
+
+The size of ONNX models can vary greatly depending on the complexity of the model and the number of parameters. They can be as small as a few KBs or as large as several GBs. While ONNX Runtime Web is designed to run all models in the browser, there are some considerations to keep in mind when working with large models.
+
+## Contents
+{: .no_toc}
+
+* TOC
+{:toc}
+
+
+## Platform restrictions
+
+There are some platform restrictions that you should be aware of when working with large models in the browser:
+
+### Maximum size of ArrayBuffer
+
+Although there is no hard limit on the size of an ArrayBuffer in JavaScript, each browser has its own limitations. For example, the maximum size of an ArrayBuffer in Chrome is 0x7fe00000 bytes (around 2GB). Be careful when using the `fetch` API to load large models, as it may fail if you call `response.arrayBuffer()` on a large file.
+
+ONNX Runtime Web creates array buffers > 2GB by using [`new WebAssembly.Memory()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/Memory/Memory) to bypass this limitation. However, an ArrayBuffer instance created by `new WebAssembly.Memory()` is not transferable, so it cannot work with the [Proxy](https://onnxruntime.ai/docs/api/js/interfaces/Env.WebAssemblyFlags.html#proxy) feature.
+
+### Protobuf file size limit
+
+The ONNX model is serialized in the protobuf format. The maximum size of a protobuf file is 2GB. If an ONNX model is larger than 2GB, it's usually generated with external data. See [External Data](#external-data) for more details.
+
+### WebAssembly memory limit
+
+WebAssembly has a memory limit of 4GB. This is the maximum amount of memory that a WebAssembly module can access because of the 32-bit addressing. Currently, there is no way for ONNX Runtime Web to run models that larger than 4GB. We may support it in the future either by using WASM64 or by using direct GPU weight loading.
```

Comment:
```suggestion
WebAssembly has a memory limit of 4GB. This is the maximum amount of memory that a WebAssembly module can access because of the 32-bit addressing. Currently, there is no way for ONNX Runtime Web to run models larger than 4GB. We may support it in the future either by using WASM64 or by using direct GPU weight loading.
```